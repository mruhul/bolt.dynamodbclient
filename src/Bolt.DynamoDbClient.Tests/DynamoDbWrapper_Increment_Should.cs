using Amazon.DynamoDBv2;
using Amazon.DynamoDBv2.Model;
using Bolt.DynamoDbClient.Tests.Builders;
using Bolt.DynamoDbClient.Tests.TestHelpers;

namespace Bolt.DynamoDbClient.Tests
{
    public class DynamoDbWrapper_Increment_Should
    {
        private IDynamoDbWrapper sut;
        private IAmazonDynamoDB fake;

        public DynamoDbWrapper_Increment_Should()
        {
            fake = Substitute.For<IAmazonDynamoDB>();
            sut = DynamoDbWrapperBuilder.Build(fake);
        }

        [Fact]
        public async void call_dynamo_with_correct_request()
        {
            UpdateItemRequest gotRequest = null;

            fake.UpdateItemAsync(Arg.Any<UpdateItemRequest>(), Arg.Any<CancellationToken>())
                .Returns(new UpdateItemResponse())
                .AndDoes(c =>
                {
                    gotRequest = c.Arg<UpdateItemRequest>();
                });


            await sut.Increment<ComplexRecord>(new IncrementRequest
            { 
                PartitionKey = "pk", 
                SortKey = "sk", 
                PropertyValues = [
                    new IncrementPropertyValue
                    {
                        PropertyName = "IntValue",
                        IncrementBy = 2
                    },
                    new IncrementPropertyValue
                    {
                        PropertyName = "Int2Value",
                        IncrementBy = -1
                    },    
                ]
            }, CancellationToken.None);

            await fake.Received().UpdateItemAsync(Arg.Any<UpdateItemRequest>(), Arg.Any<CancellationToken>());

            new
            {
                GotRequest = gotRequest
            }.ShouldMatchApproved();
        }
    }
}
