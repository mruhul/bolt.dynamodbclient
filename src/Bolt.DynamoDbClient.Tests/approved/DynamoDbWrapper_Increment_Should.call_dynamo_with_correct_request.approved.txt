{
  "gotRequest": {
    "attributeUpdates": {},
    "expected": {},
    "expressionAttributeNames": {
      "#IntValue": "IntValue"
    },
    "expressionAttributeValues": {
      ":start": {
        "bool": false,
        "isBOOLSet": false,
        "bs": [],
        "isBSSet": false,
        "l": [],
        "isLSet": false,
        "m": {},
        "isMSet": false,
        "n": "0",
        "ns": [],
        "isNSSet": false,
        "null": false,
        "ss": [],
        "isSSSet": false
      },
      ":incr": {
        "bool": false,
        "isBOOLSet": false,
        "bs": [],
        "isBSSet": false,
        "l": [],
        "isLSet": false,
        "m": {},
        "isMSet": false,
        "n": "2",
        "ns": [],
        "isNSSet": false,
        "null": false,
        "ss": [],
        "isSSSet": false
      }
    },
    "key": {
      "PK": {
        "bool": false,
        "isBOOLSet": false,
        "bs": [],
        "isBSSet": false,
        "l": [],
        "isLSet": false,
        "m": {},
        "isMSet": false,
        "ns": [],
        "isNSSet": false,
        "null": false,
        "s": "pk",
        "ss": [],
        "isSSSet": false
      },
      "SK": {
        "bool": false,
        "isBOOLSet": false,
        "bs": [],
        "isBSSet": false,
        "l": [],
        "isLSet": false,
        "m": {},
        "isMSet": false,
        "ns": [],
        "isNSSet": false,
        "null": false,
        "s": "sk",
        "ss": [],
        "isSSSet": false
      }
    },
    "tableName": "test-store",
    "updateExpression": "SET #IntValue = if_not_exists(#IntValue, :start) \u002B :incr"
  }
}